lib/persist-state.c

```diff
@@ -37,7 +37,7 @@
 
 #define PERSIST_FILE_INITIAL_SIZE 16384
 #define PERSIST_STATE_KEY_BLOCK_SIZE 4096
-#define PERSIST_FILE_WATERMARK 4096
+#define PERSIST_FILE_WATERMARK 8448
```

Comment:
this should be aligned on a 4k boundary.

Reply:
let's add an assert into the persist file allocation function that would limit the allocation to sizes that we can actually guarantee.

Reply:
Is the 4k page-boundary necessary? `PERSIST_FILE_WATERMARK` seems to be just a free-space threshold, syslog-ng always allocates `PERSIST_FILE_INITIAL_SIZE=16384` chunks, and `_grow_store()` makes sure it's page-aligned.

Reply:
Is the 4k page-boundary necessary? `PERSIST_FILE_WATERMARK` seems to be just a free-space threshold, syslog-ng always allocates `PERSIST_FILE_INITIAL_SIZE=16384` chunks, and `_grow_store()` makes sure it's page-aligned with `getpagesize()` (I think that might return 8 and 16k as well).

Reply:
Ok, that's ok this way then.

Reply:
@bazsi Thank you.